src/gpuarray_collectives_cuda_nccl.c

```diff
@@ -175,11 +175,14 @@ static inline ncclDataType_t convert_data_type(int typecode) {
   switch (typecode) {
   case GA_BYTE: return ncclChar;
   case GA_INT: return ncclInt;
-  case GA_HALF: return ncclHalf;
   case GA_FLOAT: return ncclFloat;
   case GA_DOUBLE: return ncclDouble;
   case GA_LONG: return ncclInt64;
   case GA_ULONG: return ncclUint64;
+  #ifdef CUDA_HAS_HALF
```

Comment:
Yes don't use an ifdef switch since we load the libraries dynamically and we don't want to disable it for future loads.